### DISCUSSION TITLE
Resource type naming convention

### DISCUSSION THREAD

**Discussion by kwin** [adobe/aem-core-wcm-components#1327] [General]
According to https://github.com/adobe/aem-core-wcm-components/wiki/versioning-policies the resource types always contain a version specifier, but I am wondering about why the resource type contains the actual component name part twice. 

So why did you go with `core/wcm/components/image/v2/image` instead of just simple `core/wcm/components/v2/image`? Is it just to be able to quickly see how many versions are provided for the image component (because all of them are located in the same folder) or  is there any other reason?

**Comment by vladbailescu** [comment]
@kwin , yes, the main goal was to be able to group all versions of the same component. We also toyed around with making the root a component that always points to the newest version but that proved, in our tests, to be an anti-pattern, hence recommendation of using customer-defined proxies that target the desired component version.

**Reply by puradawid** [reply]
On top of that:

if you make a root called v2, then, by the convention, main HTL renderer will be named `v2.html`.
